### PR TITLE
Convert mod_carboncopy to use maps as options

### PR DESCRIPTION
**Pull request by arcusfelis**

### CODE REVIEW

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/115941) / small_tests / 1d2c3456a8521a3fb714f87d4e9ad2279fefb4d3
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3602/115941/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3602/115941/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-03-22_08.36.19/lib.mongooseim.logs/run.2022-03-22_08.36.20/suite.log.html)

**Comment by mongoose-im** [comment]
[dynamic_domains_pgsql_mnesia_23](https://circleci.com/gh/esl/MongooseIM/115988) / pgsql_mnesia / 8c195a18216db50384e7b98c64ff6289d5cee09a
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3602/115988/pgsql_mnesia.23.1)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/116139) / small_tests / 940596cd78217b7db9cae47fdb015e7d53fedd9d
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3602/116139/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3602/116139/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-03-22_13.38.49/lib.mongooseim.logs/run.2022-03-22_13.38.49/suite.log.html)

File: src/mod_carboncopy.erl [hunk 1]
```diff
@@ -75,8 +75,8 @@ is_carbon_copy(Packet) ->
     end.
 
 start(HostType, Opts) ->
-    %% execute disable/enable actions in the c2s process itself
-    IQDisc = gen_mod:get_opt(iqdisc, Opts, no_queue),
+    %% Default no_queue - executes disable/enable actions in the c2s process itself
+    IQDisc = gen_mod:get_opt(iqdisc, Opts),
```
Comment (NelsonVides) [inline src/mod_carboncopy.erl:79]: Maybe match on the iqdisc on the start function head?

**Review by chrzaszcz — APPROVED** [review]
Looks good 👍

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/116468) / small_tests / 374bf1ff595f6a151cbaf24286b45f5f1fcd3a0a
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3602/116468/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3602/116468/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-03-23_16.17.49/lib.mongooseim.logs/run.2022-03-23_16.17.50/suite.log.html)

**Pull request merged**